src/api/scopes/helpers/add-scope-to-member.test.js

```diff
@@ -0,0 +1,254 @@
+import Boom from '@hapi/boom'
+import { addYears } from 'date-fns'
+
+import { connectToTestMongoDB } from '../../../../test-helpers/connect-to-test-mongodb.js'
+import {
+  replaceOne,
+  deleteMany
+} from '../../../../test-helpers/mongo-helpers.js'
+import {
+  userAdminFixture,
+  userAdminOtherFixture,
+  userTenantFixture,
+  memberWithGranularScopesFixture
+} from '../../../__fixtures__/users.js'
+import {
+  platformTeamFixture,
+  tenantTeamFixture
+} from '../../../__fixtures__/teams.js'
+import {
+  adminScopeFixture,
+  canGrantBreakGlassScopeFixture,
+  breakGlassScopeFixture,
+  terminalScopeFixture
+} from '../../../__fixtures__/scopes.js'
+import { addScopeToMember } from './add-scope-to-member.js'
+import { addScopeToMemberTransaction } from '../../../helpers/mongo/transactions/scope/add-scope-to-member-transaction.js'
+
+vi.mock('@azure/identity')
+
+const userCollection = 'users'
+const scopeCollection = 'scopes'
+const teamCollection = 'teams'
+const request = {}
+let replaceOneTestHelper
+
+vi.mock(
+  '../../../helpers/mongo/transactions/scope/add-scope-to-member-transaction.js'
+)
+
+const generateDates = () => {
+  const now = new Date()
+  const startDate = now
+  const endDate = addYears(now, 100)
+
+  return { startDate, endDate }
+}
+
+beforeAll(async () => {
+  vi.useFakeTimers()
+  vi.setSystemTime(new Date('2025-08-12T14:16:00.000Z'))
+
+  const { db, client } = await connectToTestMongoDB()
+  request.db = db
+  request.client = client
+
+  replaceOneTestHelper = replaceOne(db)
+})
+
+beforeEach(async () => {
+  await deleteMany(request.db)([
+    userCollection,
+    scopeCollection,
+    teamCollection
+  ])
+})
+
+describe('#addScopeToMember', () => {
+  test('Successfully adds scope with teamId to member when all conditions are met', async () => {
+    const { startDate, endDate } = generateDates()
+
+    await replaceOneTestHelper(userCollection, userAdminOtherFixture)
+    await replaceOneTestHelper(scopeCollection, canGrantBreakGlassScopeFixture)
+    await replaceOneTestHelper(teamCollection, platformTeamFixture)
+
+    await addScopeToMember({
+      request,
+      userId: userAdminOtherFixture._id,
+      scopeId: canGrantBreakGlassScopeFixture._id.toHexString(), // mimic string being passed via api endpoint
```

Comment:
This toHexString() everywhere makes me want to remove scopeIds and just use string names :-D as per teamIds